cpu/stm32/periph/i2c_2.c

```diff
@@ -205,22 +205,8 @@ static void _init(i2c_t dev)
     I2C_TypeDef *i2c = i2c_config[dev].dev;
 
     uint32_t ccr = 0;
-    /* read speed configuration */
-    switch (i2c_config[dev].speed) {
-        case I2C_SPEED_LOW:
-            /* 10Kbit/s */
-            ccr = i2c_config[dev].clk / 20000;
-            break;
-
-        case I2C_SPEED_NORMAL:
-            /* 100Kbit/s */
-            ccr = i2c_config[dev].clk / 200000;
-            break;
-
-        case I2C_SPEED_FAST:
-            ccr = i2c_config[dev].clk / 800000;
-            break;
-    }
+    /* apply speed configuration */
+    ccr = i2c_config[dev].clk / (i2c_config[dev].speed << 1);
```

Comment:
`i2c_speed_t` on stm32 is just an enum, so this won't yet work

Reply:
Why not? `enum`s are just "fancy integers" in C. And in any case `speed` should be converted to `int` or `unsigned` due to integer promotion anyway, shouldn't it?

But I could add an explicit cast, if you prefer.

Reply:
You didn't change the enum, it's still

```C
typedef enum {
#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || \
    defined(CPU_FAM_STM32F4) || defined(CPU_FAM_STM32L1) || \
    defined(CPU_FAM_STM32MP1)
    I2C_SPEED_LOW,          /**< low speed mode: ~10kit/s */
#endif
    I2C_SPEED_NORMAL,       /**< normal mode:  ~100kbit/s */
    I2C_SPEED_FAST,         /**< fast mode:    ~400kbit/s */
#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F3) || \
    defined(CPU_FAM_STM32F7) || defined(CPU_FAM_STM32L0) || \
    defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32WB) || \
    defined(CPU_FAM_STM32G4) || defined(CPU_FAM_STM32G0) || \
    defined(CPU_FAM_STM32L5) || defined(CPU_FAM_STM32U5) || \
    defined(CPU_FAM_STM32WL)
    I2C_SPEED_FAST_PLUS,    /**< fast plus mode: ~1Mbit/s */
#endif
} i2c_speed_t;
```

so `I2C_SPEED_LOW` is just `0`

Reply:
Oops :-)

Since this got more involved now, I split this change out to https://github.com/RIOT-OS/RIOT/pull/20169